heat/spatial/distance.py

```diff
@@ -363,43 +360,27 @@ def _dist(X: DNDarray, Y: DNDarray = None, metric: Callable = _euclidian) -> DND
 
         else:
             raise NotImplementedError(
-                "Input split was X.split = {}. Splittings other than 0 or None currently not supported.".format(
-                    X.split
-                )
+                f"Input split was X.split = {X.split}. Splittings other than 0 or None currently not supported."
             )
-    # Y is not None
     else:
         if len(Y.shape) > 2:
             raise NotImplementedError(
-                "Only 2D data matrices are supported, but input shapes were X: {}, Y: {}".format(
-                    X.shape, Y.shape
-                )
+                f"Only 2D data matrices are supported, but input shapes were X: {X.shape}, Y: {Y.shape}"
             )
 
         if X.comm != Y.comm:
             raise NotImplementedError("Differing communicators not supported")
 
-        if X.split is None:
-            if Y.split is None:
-                split = None
-            elif Y.split == 0:
-                split = 1
-            else:
-                raise NotImplementedError(
-                    "Input splits were X.split = {}, Y.split = {}. Splittings other than 0 or None currently not supported.".format(
-                        X.split, Y.split
-                    )
-                )
-        elif X.split == 0:
-            split = X.split
-        else:
-            # ToDo: Find out if even possible
+        if X.split is None and Y.split is None:
+            split = None
+        elif X.split is None and Y.split == 0:
+            split = 1
+        elif X.split is None or X.split != 0:
```

Comment:
I'm not sure we should merge conditional statements here as we end up checking if X.split is None three times before moving on to the next case.

Reply:
Yeah, I think it doesn't matter much but I will change it back to the original one.